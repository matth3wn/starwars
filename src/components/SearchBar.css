input[type="text"] {
  width: 200px;
  height: 30px;
  font-size: 16px;
  margin: 5px;
}

select{
  height: 30px;
  font-size: 16px;
  margin: 5px;
  font: Franklin Gothic;
}

button{
  height: 30px;
  font-size: 16px;
  margin: 5px;
  font: Franklin Gothic;
}
li{
  list-style-type: none;
  padding: 2%;
  font: Franklin Gothic;
}