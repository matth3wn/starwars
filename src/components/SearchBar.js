import React, { Component } from 'react';
import './SearchBar.css';

class SearchBar extends Component {
  constructor(props) {
    super(props);
    this.state = {
      searchTerm: '',
      options: null,
    }
  }
  
  handleChange(term) {
    this.setState({
      searchTerm: term
    });
  }

  handleSelct(option){
    this.setState({
      options: option
    });
  }

  handleSubmit(e) {
    e.preventDefault();

    if(this.state.options === ''){
      this.setState({
        option:true,
      })
    }
    else{
      this.props.search(this.state.searchTerm, this.state.options);
    }
    
  }

  render() {
    return (
      <div>
        <form className="searchbar" onSubmit={(e) => this.handleSubmit(e)}>
          <input
            type="text"
            name="search"
            id="search"
            className='search'
            onChange={(e) => this.handleChange(e.target.value)}
          />
          <select onChange={(e)=>this.handleSelct(e.target.value) }>
           <option value= ''>--Select--</option>
            <option value="planets">planets</option>
            <option value="starships">spaceships</option>
            <option value="vehicles">vehicles</option>
            <option value="people">characters</option>
            <option value="films">films</option>
            <option value="species">species</option>
          </select>
          <button className='submit' type="submit"> Search!</button>
        </form>
        {this.state.options && <p>Please select an option...</p>}
      </div>

    );
  }
}

export default SearchBar;