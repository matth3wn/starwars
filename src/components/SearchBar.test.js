import React from 'react'
import ReactDOM from 'react-dom'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import SearchBar from './SearchBar'

describe(`SearchBar Component`, () => {

  it('renders without errors', () => {
    const div = document.createElement('div');
    ReactDOM.render(<SearchBar search={(e, c) => this.handleSearch(e, c)} />, div);
    ReactDOM.unmountComponentAtNode(div);
  })

  it('renders searchbar', () => {
    const wrapper = shallow(<SearchBar  />)
    wrapper.find('button').at(0).simulate('submit')
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
