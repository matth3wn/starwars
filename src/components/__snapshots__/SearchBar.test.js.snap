// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar Component renders searchbar 1`] = `
<div>
  <form
    className="searchbar"
    onSubmit={[Function]}
  >
    <input
      className="search"
      id="search"
      name="search"
      onChange={[Function]}
      type="text"
    />
    <select
      onChange={[Function]}
    >
      <option
        value=""
      >
        --Select--
      </option>
      <option
        value="planets"
      >
        planets
      </option>
      <option
        value="starships"
      >
        spaceships
      </option>
      <option
        value="vehicles"
      >
        vehicles
      </option>
      <option
        value="people"
      >
        characters
      </option>
      <option
        value="films"
      >
        films
      </option>
      <option
        value="species"
      >
        species
      </option>
    </select>
    <button
      className="submit"
      type="submit"
    >
       Pew! Pew!
    </button>
  </form>
  <li>
     Please select an option  
  </li>
</div>
`;
